pkg/agent/cniserver/pod_configuration_test.go

```diff
@@ -0,0 +1,461 @@
+		},
+	)
+	podLister := corelisters.NewPodLister(localPodInformer.GetIndexer())
+	ofClient := openflowtest.NewMockClient(ctrl)
+	recorder := record.NewFakeRecorder(100)
+	recorder.IncludeObject = false
+
+	return &mockClients{
+		kubeClient:       kubeClient,
+		informerFactory:  informerFactory,
+		localPodInformer: localPodInformer,
+		podLister:        podLister,
+		ofClient:         ofClient,
+		recorder:         recorder,
+	}
+}
+
+func (c *mockClients) startInformers(stopCh chan struct{}) {
+	c.informerFactory.Start(stopCh)
+	c.informerFactory.WaitForCacheSync(stopCh)
+	go c.localPodInformer.Run(stopCh)
+	cache.WaitForCacheSync(stopCh, c.localPodInformer.HasSynced)
+}
+
+type asyncWaiter struct {
+	podName     string
+	containerID string
+	waitCh      chan struct{}
+	stopCh      chan struct{}
+	notifier    *channel.SubscribableChannel
+}
+
+func (w *asyncWaiter) notify(e interface{}) {
+	podUpdate := e.(types.PodUpdate)
+	if podUpdate.PodName == w.podName && podUpdate.ContainerID == w.containerID {
+		w.waitCh <- struct{}{}
+	}
+}
+
+func (w *asyncWaiter) wait() {
+	<-w.waitCh
+}
+
+func (w *asyncWaiter) close() {
+	close(w.waitCh)
+	close(w.stopCh)
+}
+
+func newAsyncWaiter(podName, containerID string) *asyncWaiter {
+	waiter := &asyncWaiter{
+		podName:     podName,
+		containerID: containerID,
+		waitCh:      make(chan struct{}),
+		stopCh:      make(chan struct{}),
+		notifier:    channel.NewSubscribableChannel("PodUpdate", 100),
+	}
+	waiter.notifier.Subscribe(waiter.notify)
+	go waiter.notifier.Run(waiter.stopCh)
+	return waiter
+}
+
+func mockRetryInterval() func() {
+	oriRetryInterval := retryInterval
+	retryInterval = time.Millisecond * 500
+	return func() {
+		retryInterval = oriRetryInterval
+	}
+}
+
+func assertEqualEvents(t *testing.T, expected string, actual <-chan string) {
+	c := time.After(wait.ForeverTestTimeout)
+	select {
+	case a := <-actual:
+		require.Equal(t, expected, a)
+	case <-c:
+		t.Errorf("Expected event %q, got nothing", expected)
+		// continue iterating to print all expected events
+	}
+}
+
+func newTestPodConfigurator(testClients *mockClients, waiter *asyncWaiter) *podConfigurator {
+	interfaceStore := interfacestore.NewInterfaceStore()
+	eventBroadcaster := record.NewBroadcaster()
+	queue := workqueue.NewTypedDelayingQueueWithConfig[string](
+		workqueue.TypedDelayingQueueConfig[string]{
+			Name: "podConfigurator",
+		},
+	)
+	podCfg := &podConfigurator{
+		kubeClient:       testClients.kubeClient,
+		ofClient:         testClients.ofClient,
+		podLister:        testClients.podLister,
+		ifaceStore:       interfaceStore,
+		eventBroadcaster: eventBroadcaster,
+		recorder:         testClients.recorder,
+		unreadyPortQueue: queue,
+		containerAccess:  newContainerAccessArbitrator(),
+	}
+	if waiter != nil {
+		podCfg.podUpdateNotifier = waiter.notifier
+	}
+	return podCfg
+}
+
+func TestUpdateUnreadyPod(t *testing.T) {
+	defer mockRetryInterval()()
```

Comment:
probably better to pass `t` to `mockRetryInterval` and use the `t.Cleanup` support